Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
+        }
+      }
+      catch (const MolSanitizeException &){
+        std::cerr << "Molecule could not be sanitized" << std::endl;
+      }
+    }
+  }
+  else{
+    mol=SmilesToMol(txt,0,sanitize,replacements);
+  }
+  return mol;
+}
+
+
+
+
 }  // end of namespace DaylightParserUtils
 
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
-    bool useSmiles) {
+    bool useSmiles, bool sanitize) {
```

Comment:
I think you guys have different goals.
Brian wants to sanitize the reactions to make them more useful as reactions.
You want to get the appropriate computed properties to get decent pictures of the molecules.
How about changing the name of the argument in your function to `sanitizeComponents`? That would make it clear that it's not dealing with the whole reaction.